### ISSUE TITLE
Update message preview in room list in regards to Thread

### ISSUE THREAD

**Issue by gsouquet** [vector-im/element-web#19615]
Behaviour to be defined by @janogarcia 

How does the message preview in the room list works in regards to threads. Does the preview show all new messages, or just the ones from the main timeline?

<img width="330" alt="Screen Shot 2021-11-30 at 13 06 35" src="https://user-images.githubusercontent.com/60790/144053038-2afe3a5d-cbbf-4dc0-ae25-6b01f6d7c29d.png">

**Comment by janogarcia** [comment]
@gsouquet Unsure what this issue is about. Could you please add context?

**Comment by janogarcia** [comment]
@gsouquet I've edited the issue description with the screenshot you provided me via DM. 👍

**Comment by gsouquet** [comment]
@janogarcia Could you update this? 
Not entirely sure how we want to handle this

**Comment by janogarcia** [comment]
@gsouquet  Sure, I worked on those before leaving for vacation, you can find the relevant mockups and specs here:

**Component: Room list**
https://www.figma.com/file/T309ztx0sNyOOK6NKVLHsK/Threads?node-id=2730%3A263784

/cc @daniellekirkwood 

---

**Snapshot**

Only threads that match the same conditions for the “My threads” filter can generate a preview in the room list. A thread reply preview can be shown as long as the latest message in the main timeline is older.

If that’s not easily doable, we can just omit thread reply previews for the Threads P0 Milestone.

![Threads – Figma](https://user-images.githubusercontent.com/60790/145863487-66011d8a-a1d5-40eb-8e4e-427ce0104849.png)

**Comment by daniellekirkwood** [comment]
I agree with Jano's appraoch. Including "My Threads" info would be great, but let's see what @gsouquet says about how heavy a lift this is for P0.

Also, do Mobile apps have previews?

**Comment by janogarcia** [comment]
I tried to keep a balance between giving visibility to relevant updates across the room (main timeline and threads you are participating in) and overwhelming the user. That's why we are not showing previews for just about any thread in the room (those that you're not participating in).

Plus, that last case could be more aptly addressed with the [Thread Events](https://www.figma.com/file/T309ztx0sNyOOK6NKVLHsK/Threads?node-id=1248%3A231503) proposal, which could generate a preview in the room list, as those would be posted to the room timeline.

> Also, do Mobile apps have previews?

Yes, they do. I'll work on similar proposals for iOS/Android.

**Comment by gsouquet** [comment]
This might be possible under the condition that we can determine from the `unsigned` field whether the logged in user has participated in a thread or not.
Using the result from the filtered messages API calls would probably be too complex for this. 

@janogarcia what happens when you click on the room and the previewed messages is in a thread? Should we proactively open that thread? I feel like it could create quite a jarring experience if folks open that bit and end up not seeing the message they came for

**Comment by gsouquet** [comment]
Blocked by https://github.com/matrix-org/synapse/issues/11399

**Comment by gsouquet** [comment]
To test against https://github.com/matrix-org/synapse/pull/11577

**Comment by gsouquet** [comment]
@janogarcia heads up, we are not going to be able to do this one as the backend is probably not going to be deployed on time.

But we should be able to have a follow-up to update this. In the interim we have two solutions

1. We keep the message preview as is. It only displays messages from the main timeline
2. We display ALL messages, from the main timeline to every threads regardless of your participation

**Comment by janogarcia** [comment]
@gsouquet Let's just go with `1. We keep the message preview as is. It only displays messages from the main timeline` for now, and then plan to properly handle this as a follow-up action, after the MVP:

> We can always revisit this decision and explore other, more advanced, implementations where a separate/dedicated action is available for threads in the room list (e.g. Discord's active threads tree/popover in channel list), but that's out of scope for the P0 milestone.

I've actually [explored that idea in detail on Figma](https://www.figma.com/file/T309ztx0sNyOOK6NKVLHsK/Threads?node-id=3586%3A328211) and filed it as a [post-MVP improvement](https://github.com/vector-im/element-meta/issues/61#issuecomment-1024502902).

**Comment by t3chguy** [comment]
One (major imo) issue with `1.` is now the room unread count and message previews will fall out of sync, you'll have an unread in a thread but the message preview doesn't show it, it feels broken and then finding the unread is really difficult.

**Comment by janogarcia** [comment]
@t3chguy The thing is that there isn't much we can do until we have proper backend support. The explorations I shared on Figma can potentially deal with all those scenarios more elegantly, effectively separating the thread-related updates from those happening in the main timeline but still in a visually compact way.

**Partial screenshot — WIP**

![Threads – Figma](https://user-images.githubusercontent.com/60790/151861719-de47a477-13f1-4a98-97f3-01e41a23ff65.png)

As for the current workarounds, `1.` vs `2.`, none is ideal, both have their fair share of shortcomings. For example, for `2.` you would still experience the same issue you described for `1.` as soon as the latest message in the room is posted to the main timeline, so there's really not much benefit to it, apart from being noisier. That's why I think it's not worth the effort.

**Comment by gsouquet** [comment]
Closing in favour of https://github.com/vector-im/element-web/issues/23920 that explains all the requirements for this feature to work.
